proof/refine/RISCV64/ADT_H.thy

```diff
@@ -405,15 +444,28 @@ lemma distinct_word_add_inj_ptes:
    \<Longrightarrow> p' = p \<and> off' = off" for off :: pt_index and p :: machine_word
   by (erule (2) distinct_word_add_ucast_shift_inj; simp add: bit_simps)
 
+context
+begin
+
+private method ako =
+  find_goal \<open>match premises in "kheap s p = Some (ArchObj ako)" for s p ako \<Rightarrow> succeed\<close>,
+  (rename_tac ako, case_tac ako;
+   clarsimp simp: other_obj_relation_def pte_relation_def split: if_split_asm)
+
+private method ko =
+  case_tac ko; clarsimp simp: other_obj_relation_def cte_relation_def split: if_split_asm
```

Comment:
Nice, this might be worth back-porting to master for AArch64 when we get to Refine there.